cli/src/pcluster/config/update_policy.py

```diff
@@ -125,6 +134,54 @@ def is_slurm_queues_change(change):
     return any(path.startswith("SlurmQueues[") for path in change.path)
 
 
+def extract_type_and_name_from_path(path):
+    obj_type = re.search(".+(?=\\[)", path).group(0)
+    obj_name = re.search("(?<=\\[).+(?=\\])", path).group(0)
+    return obj_type, obj_name
+
+
+def get_q_from_config(change, config):
+    q_type, q_name = None, None
+    for path in change.path:
+        if re.search("Queues\\[", path):
+            q_type, q_name = extract_type_and_name_from_path(path)
+    if q_type and q_name:
+        for queue in config.get("Scheduling", {}).get(q_type, {}):
+            if queue.get("Name", None) == q_name:
+                return queue
+    return {}
+
+
+def get_cr_from_config(change, config):
```

Comment:
Let's add a sample string of the `change` to better understand how the regex parsing?/matching is happening.

Reply:
will do!